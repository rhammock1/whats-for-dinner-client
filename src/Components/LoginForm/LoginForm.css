.login-form {
  width: 75%;
  height: auto;
  margin: auto;
  padding: 5px 10px;
  background-color: #313753;
  border-radius: 5px;
  box-shadow: 2px 3px 6px 3px #121212; 
}
input {
  padding: 5px;
}
.sign-up {
  background-color: #434D77;
  border-radius: 5px;
  box-shadow: 2px 3px 6px 3px #121212; 
  padding:2px;
  margin-top: 10px;
}
.sign-up>p {
  margin: 0;
  padding: 5px;
}
p>a {
  color: #efefef
}