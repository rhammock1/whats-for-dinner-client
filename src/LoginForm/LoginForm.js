import React from 'react';
import TokenService from '../services/token-service';
import AuthApiService from '../services/auth-api-service';
import {Link} from 'react-router-dom';

class LoginForm extends React.Component {
  static defaultProps = {
    onLoginSuccess: () => {}
  }

  state = { error: null }

  handleSubmitJWTAuth = event => {
    event.preventDefault();
    this.setState({ error: null })
    const { user_name, password } = event.target;

    AuthApiService.postLogin({
      user_name: user_name.value,
      password: password.value,
    })
      .then(res => {
        user_name.value = '';
        password.value = '';
        TokenService.saveAuthToken(res.authToken)
        TokenService.saveUserName(res.user_name)
        this.props.onLoginSuccess();
        
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }
  render() {
    const { error } = this.state;
    return (
      <form className='login-form' onSubmit={this.handleSubmitJWTAuth}>
        <fieldset>
          <legend>Form</legend>
          <div role='alert'>{error && <p className='red'>{error}</p>}</div>
          <div className='form-group'>
            <label htmlFor='user_name'>Username: </label>
            <input type='text' required id='user_name' name='user_name' />
          </div>
          <div className='form-group'>
            <label htmlFor='password'>Password: </label>
            <input type='password' required id='password' name='password' />
          </div>
          <button type='submit'>Login</button>
          <p>Need an account? <br/><Link to='/register'>Sign up</Link></p>
        </fieldset>
      </form>
    )
  }
}

export default LoginForm;