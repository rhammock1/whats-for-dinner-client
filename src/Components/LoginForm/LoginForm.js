/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
import { Link } from 'react-router-dom';
import TokenService from '../../services/token-service';
import AuthApiService from '../../services/auth-api-service';
import Context from '../../Context';
import './LoginForm.css';

class LoginForm extends React.Component {
  static defaultProps = {
    onLoginSuccess: () => {},
  }

  static contextType = Context;

  state = { error: null }

  handleSubmitJWTAuth = (event) => {
    event.preventDefault();
    this.setState({ error: null });
    const { user_name, password } = event.target;

    AuthApiService.postLogin({
      user_name: user_name.value,
      password: password.value,
    })
      .then((res) => {
        user_name.value = '';
        password.value = '';
        TokenService.saveAuthToken(res.authToken);
        TokenService.saveUserName(res.user_name);
        TokenService.saveUserId(res.id);
        const { onLoginSuccess } = this.props;
        onLoginSuccess();
      })
      .catch((res) => {
        this.setState({ error: res.error });
      });
  }

  render() {
    const { error } = this.state;
    return (
      <form className="login-form" onSubmit={this.handleSubmitJWTAuth}>
        <fieldset>
          <legend><h3>Login Credentials</h3></legend>
          <div role="alert">{error && <p className="red">{error}</p>}</div>
          <div className="form-group">
            <label htmlFor="user_name">Username: </label>
            <input type="text" required id="user_name" name="user_name" />
          </div>
          <div className="form-group">
            <label htmlFor="password">Password: </label>
            <input type="password" required id="password" name="password" />
          </div>
          <button type="submit">Login</button>
          <div className="sign-up">
            <p>
              Need an account?
              <br />
              <Link to="/register">Sign up</Link>
            </p>
          </div>
        </fieldset>
      </form>
    );
  }
}

export default LoginForm;
