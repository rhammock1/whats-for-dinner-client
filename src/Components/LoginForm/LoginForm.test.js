import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LoginForm from './LoginForm';


describe('Login Form ', () => {
  it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<BrowserRouter><LoginForm />
  </BrowserRouter>, div)
  })
  it('renders the complete form', () => {
    const wrapper = shallow(<LoginForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})