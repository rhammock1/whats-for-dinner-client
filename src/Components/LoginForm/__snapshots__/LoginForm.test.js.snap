// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form  renders the complete form 1`] = `
<form
  className="login-form"
  onSubmit={[Function]}
>
  <fieldset>
    <legend>
      <h3>
        Login Credentials
      </h3>
    </legend>
    <div
      role="alert"
    />
    <div
      className="form-group"
    >
      <label
        htmlFor="user_name"
      >
        Username: 
      </label>
      <input
        id="user_name"
        name="user_name"
        required={true}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="password"
      >
        Password: 
      </label>
      <input
        id="password"
        name="password"
        required={true}
        type="password"
      />
    </div>
    <button
      type="submit"
    >
      Login
    </button>
    <div
      className="sign-up"
    >
      <p>
        Need an account?
        <br />
        <Link
          to="/register"
        >
          Sign up
        </Link>
      </p>
    </div>
  </fieldset>
</form>
`;
